brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx

```diff
@@ -1,39 +1,45 @@
 {% extends 'common.pyx' %}
 
-{% block template_support_code %}
-cdef int _start_idx = 0
-cdef double _current_time = 0.0
-{% endblock %}
-
 {% block maincode %}
 
-    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time } #}
-     
-    global _start_idx, _current_time
-    
-    # Check if we have been reset, if so reset the cache
-    if t<_current_time:
-        _start_idx = 0
-    
-    _current_time = t
+    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time, period, _lastindex } #}
 
     # TODO: We don't deal with more than one spike per neuron yet
-    cdef int _cpp_numspikes = 0
+    cdef int _cpp_numspikes   = 0
+    cdef float padding_before = t % period
+    cdef float padding_after  = (t + dt) % period
+    cdef float epsilon        = 1e-3*dt

```

Comment:
Actually, this implementation without the logarithmic search has never been
benchmarked, but I always assumed that this would be faster. It "felt" as
if it was simpler.... If this is not the case, then this is problematic...
I see no reason why the used should change epsilon, and this can (should)
be set to the smallest possible value giving correct results for multiples
of dt.

We can talk about that tomorrow

2015-01-21 15:20 GMT+01:00 Marcel Stimberg <notifications@github.com>:

> In brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx
> <https://github.com/brian-team/brian2/pull/339#discussion_r23301527>:
>
> >
> >      # TODO: We don't deal with more than one spike per neuron yet
> > -    cdef int _cpp_numspikes = 0
> > +    cdef int _cpp_numspikes   = 0
> > +    cdef float padding_before = t % period
> > +    cdef float padding_after  = (t + dt) % period
> > +    cdef float epsilon        = 1e-3*dt
>
> It's a bit of an opaque constant, I'm not sure this is something the
> user'd like to set. Is there any reason for not setting it to the smallest
> epsilon that still gives correct results for multiples of dt?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brian-team/brian2/pull/339/files#r23301527>.
>